specs/online_order_spec.rb

```diff
@@ -1,90 +1,133 @@
 # only test things that are different.
 
-xdescribe "OnlineOrder" do
+describe "OnlineOrder" do
+
   describe "#initialize" do
+
+    before do
+      @online_order = Grocery::OnlineOrder.new(4, {"Aubergine" => 56.71, "Brown rice vinegar" => 33.52, "dried Chinese Broccoli" => 51.74}, 35, "paid")
+    end
+
     it "Is a kind of Order" do
       # Check that an OnlineOrder is in fact a kind of Order
-
-      # Instatiate your OnlineOrder here
-      # online_order =
-      # online_order.must_be_kind_of Grocery::Order
+      @online_order.must_be_kind_of Grocery::Order
     end
 
-    it "Can access Customer object" do
+    xit "Can access Customer object" do
       # TODO: Your test code here!
     end
 
     it "Can access the online order status" do
-      # TODO: Your test code here!
+      @online_order.status.must_equal "paid"
     end
   end
 
   describe "#total" do
+
+    before do
+      @online_order = Grocery::OnlineOrder.new(4, {"Aubergine" => 56.71, "Brown rice vinegar" => 33.52, "dried Chinese Broccoli" => 51.74}, 35, "paid")
+    end
+
     it "Adds a shipping fee" do
-      # TODO: Your test code here!
+      @online_order.total.must_equal 162.62
     end
 
     it "Doesn't add a shipping fee if there are no products" do
-      # TODO: Your test code here!
+      empty_online_order = Grocery::OnlineOrder.new(4, {}, 35, "paid")
+      empty_online_order.total.must_equal 0
     end
   end
 
   describe "#add_product" do
     it "Does not permit action for processing, shipped or completed statuses" do
-      # TODO: Your test code here!
+      shipped_online_order = Grocery::OnlineOrder.new(5,{"Sour Dough Bread" => 35.11}, 21, "shipped")
+      proc { shipped_online_order.add_product("bananas", 1.99) }.must_raise ArgumentError
     end
 
     it "Permits action for pending and paid satuses" do
-      # TODO: Your test code here!
+      another_order = Grocery::OnlineOrder.new(36, {"Oranges" => 12.05, "Tomatoes" => 28.46, "Mahi mahi" => 67.78}, 20, :pending)
+      another_order.add_product("bananas", 1.99).must_equal true
+      proc { another_order.add_product("bananas", 1.99) }.must_be_silent
     end
   end
 
   describe "OnlineOrder.all" do
+
+    before do
+      @@all_online_orders = Grocery::OnlineOrder.all
+    end
+
     it "Returns an array of all online orders" do
-      # TODO: Your test code here!
+      @@all_online_orders.length.must_equal 100
+      @@all_online_orders.must_be_kind_of Array
     end
 
     it "Returns accurate information about the first online order" do
-      # TODO: Your test code here!
+      first_row_online_products = {"Lobster" => "17.18", "Annatto seed" => "58.38", "Camomile" => "83.21"}
+
+      Grocery::OnlineOrder.all.first.id.must_equal 1
+      Grocery::OnlineOrder.all.first.products.must_equal first_row_online_products
+      Grocery::OnlineOrder.all.first.customer_id.must_equal 25
+      Grocery::OnlineOrder.all.first.status.must_equal :complete
```

Comment:
Every time you have `Grocery::OnlineOrder.all.first`, you're calling and invoking and running `Grocery::OnlineOrder.all`. Which isn't a bad thing!

But it would be all the same/have less code if you just assigned the result of `Grocery::OnlineOrder.all` to a local variable at the top of this method:

```
order = Grocery::OnlineOrder.all.first
order.id.must_equal 1
order.products.must_equal first_row_online_products
order.customer_id.must_equal 25
order.status.must_equal :complete
```